source/simulator/introspection.cc

```diff
@@ -332,6 +332,39 @@ namespace aspect
 
 
 
+  template <int dim>
+  std::vector<unsigned int>
+  Introspection<dim>::get_compositional_field_base_element_indices() const
+  {
+    // We are assigning base elements in order, so the first compositional field
+    // gives us the first base element index. Then we find the largest index
+    // in the vector. This is necessary, because the fields could have type A,B,A.
+    std::vector<unsigned int> result;
+    const unsigned int first = this->base_elements.compositional_fields[0];
+    const unsigned int last = *std::max_element(this->base_elements.compositional_fields.begin(), this->base_elements.compositional_fields.end());
+    for (unsigned int i = first; i<=last; ++i)
+      result.emplace_back(i);
+    return result;
+  }
```

Comment:
I thought that the function returns the base elements for all compositional fields. Then this array would have size `# of compositional fields`, but it clearly does not here.

Reply:
no, it returns a list of base element indices.

Reply:
I was under the same impression as @bangerth from reading the documentation in the header.